wikimedia/wikipedia-ios
# New fundraising banner and in-app webview donation page

**Phabricator:** https://phabricator.wikimedia.org/T333108 & https://phabricator.wikimedia.org/T346823

### Notes
*  This PR updates the UI and actions on our fundraising modal - now the user can choose to be reminded at a later date
* It also keeps the donation flow within the app, opening the pages from an in-app webview
* This PR depends on #4641

### Test Steps
1. Run the experimental app, open it once, background it, and then open it again
2. Set your region to the Netherlands - I was hard-coding "NL" on all methods that verify the country code
3. Add Dutch to your languages
4. Test closing using the "X" button or the "already donated" button (the last one) - it should not show the modal again
5. Delete the app, reinstall, repeat step 1
6. Select maybe later. Check if the reminder toast shows.
7. Visit the article again. it should not show the modal. 
8. Change the date on your phone to tomorrow. Visit the article, it should show the modal.
9. Reinstall and uninstall, repeat step 1. This time, donate.
10. Verify that the `back to article` button shows on the bottom of the Thank you page.
11. Close the web view and verify that the thank you toast shows
